### PR TITLE
Add and fix smoke tests on prove and verify

### PR DESCRIPTION
**Pull request by yoichi-nexus**
This MR aims at preventing regression of https://github.com/nexus-xyz/nexus-zkvm/pull/213 , where a newly added constraint broke the examples mentioned in README.md.

This MR
- adds a shell script `smoke.sh` that runs `cargo-nexus nexus {run,prove,verify}`
- adds two lightweight examples in `examples/src/bin` that detect the issue fixed in https://github.com/nexus-xyz/nexus-zkvm/pull/213
- adds the new examples in the CI
- fixes the proving implementation so that the new smoke tests pass

### CODE REVIEW

File: assets/scripts/smoke.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+cargo build --release --package nexus-tools --bin cargo-nexus
+./target/release/cargo-nexus nexus new $PROJECT_NAME
+cp $1 $PROJECT_NAME/src/main.rs
+cd $PROJECT_NAME
+../target/release/cargo-nexus nexus run
+../target/release/cargo-nexus nexus prove
+../target/release/cargo-nexus nexus verify
+cd $ORIGINAL_DIR
+rm -rf $PROJECT_NAME
```
Comment (yoichi-nexus) [inline assets/scripts/smoke.sh:53]: the `move -f` line like line 40 is missing.

File: assets/scripts/smoke.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+fi
+
+if [ ! -f "$1" ]; then
+    echo "Usage: $0 <file.rs>"
+    echo "where <file.rs> is a path to a file."
+    exit 1
+fi
+
+if [ -e $PROJECT_NAME ]; then
+    echo "Error: Directory 'nexus-project-ci' already exists."
```
[the remark this answers is not in the document]
Comment (yoichi-nexus) [reply assets/scripts/smoke.sh]: Fixed. Thank you.

File: assets/scripts/smoke.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+cargo build --release --package nexus-tools --bin cargo-nexus
+./target/release/cargo-nexus nexus new $PROJECT_NAME
+cp $1 $PROJECT_NAME/src/main.rs
+cd $PROJECT_NAME
+../target/release/cargo-nexus nexus run
+../target/release/cargo-nexus nexus prove
+../target/release/cargo-nexus nexus verify
+cd $ORIGINAL_DIR
+rm -rf $PROJECT_NAME
```
Comment (nexodon) [inline assets/scripts/smoke.sh:53]: DRY - could have single cleanup() method when done (whether error or success)

File: vm/src/trace.rs [hunk 4]
```diff
@@ -236,7 +236,7 @@ impl<P: MemoryProof> Iterator for BlockIter<'_, P> {
 
         let s = &self.block.steps[self.index];
         let inst = parse_u32(s.inst).unwrap();
-        let mut w = parse_alt(&self.block.regs, s.inst);
+        let mut w = parse_alt(&self.regs, s.inst);
```
Comment (sjudson) [inline vm/src/trace.rs:239]: Why this change?

File: assets/scripts/smoke.sh (hunk 3 above)
Comment (yoichi-nexus) [reply assets/scripts/smoke.sh:53]: true

File: vm/src/trace.rs (hunk 4 above)
Comment (yoichi-nexus) [reply vm/src/trace.rs:239]: self.block.regs is the initial register file for the folding block. It's immutable for the whole k-step execution. Here we need the register file that changes instruction-to-instruction.

Comment (yoichi-nexus) [reply vm/src/trace.rs:239]: Moreover, without the change, the smoke tests fail.

**Review by sjudson — APPROVED** [review]

File: assets/scripts/smoke.sh (hunk 3 above)
Comment (yoichi-nexus) [reply assets/scripts/smoke.sh]: moved those common lines into `cleanup()` function.
